lib/Service/LockService.php

```diff
@@ -137,10 +137,13 @@ public function getLockForNodeIds(array $nodeIds): array {
 		if (count($locksToRequest) === 0) {
 			return $locks;
 		}
-		$newLocks = $this->locksRequest->getFromFileIds($locksToRequest);
-		foreach ($locksToRequest as $lockId) {
-			$this->lockCache[$lockId] = false;
+
+		$newLocks = [];
+		while ($fileIds = array_splice($locksToRequest, 0, 1000)) {
+			$newLocks[] = $this->locksRequest->getFromFileIds($fileIds);
 		}
+		$newLocks = array_merge(...$newLocks);
```

Comment:
What happened to `$this->lockCache[$lockId] = false;`?

Reply:
This is overwritten when iterating over the new locks again anways below.